### ISSUE TITLE
transforms names of unrelated resource types in RBAC rules

### ISSUE THREAD

**Issue by cjyar** [kubernetes-sigs/kustomize#4366]
**Describe the bug**

Kustomize transforms names of resources in Roles without regard to the types of those resources.

**Files that can reproduce the issue**

kustomization.yaml

```
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
- resources.yaml

configMapGenerator:
- name: test
```

resources.yaml

```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: test
rules:
- apiGroups:
  - apps
  resources:
  - deployments
  verbs:
  - get
  resourceNames:
  - test
```

**Expected output**

```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: test
rules:
- apiGroups:
  - apps
  resourceNames:
  - test
  resources:
  - deployments
  verbs:
  - get
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-6ct58987ht
```

**Actual output**

```
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  name: test
rules:
- apiGroups:
  - apps
  resourceNames:
  - test-6ct58987ht
  resources:
  - deployments
  verbs:
  - get
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: test-6ct58987ht
```

**Kustomize version**: 4.4.0

**Platform**: macOS

**Comment by Serializator** [comment]
I believe this is a duplicate of #4254 which was closed in favor of #3418.

//cc @natasha41575

**Comment by natasha41575** [comment]
Yes, I am closing this for the same reason as #4254. I am currently working on a design for #3418 which will hopefully address this issue.

/close
